src/App.tsx

```diff
@@ -102,16 +103,25 @@ const rememberFilters = (filtersToRemember: FilterOptions) => {
   }
 };
 
-const initialiseFilters = (
-  pageSizeOverride?: PageSizeType,
-  orderByOverride?: OrderByType
-) => {
+const initialiseFilters = ({
+  pageSizeOverride,
+  orderByOverride,
+  permalinkBeingUsed
+}: {
+  pageSizeOverride?: PageSizeType;
+  orderByOverride?: OrderByType;
+  permalinkBeingUsed: boolean;
+}) => {
   const initialisedFilters = initFiltersFromLocalStorage();
   return {
     ...initialisedFilters,
     // Override if prop given
     pageSize: pageSizeOverride || initialisedFilters.pageSize,
-    orderBy: orderByOverride || initialisedFilters.orderBy
+    orderBy: orderByOverride || initialisedFilters.orderBy,
+    threads:
+      initialisedFilters.threads === "collapsed" && permalinkBeingUsed
```

Comment:
wouldn't this  need to be wrapped like : `(initialisedFilters.threads === "collapsed" && permalinkBeingUsed)` to avoid false being set?

Reply:
I think because it's using the '?' ternary it will alway return one of the 2 options and `initialisedFilters.threads` can't be undefined so we should be good.

Reply:
my understanding is if: `initialisedFilters.threads === "collapsed" ` is false, then  `&& permalinkBeingUsed` is never called, therefore setting `threads` to false as it never reaches the `?`. but wrapping it with `( ... )` forces the program to evaluate it as 1 condition

Reply:
Ternaries always return one or other of their outcomes so if threads is false it will jump to the negative outcome

![Screenshot 2020-03-26 at 12 20 58](https://user-images.githubusercontent.com/1336821/77646441-75aa7080-6f5c-11ea-8c5f-40d06013301e.jpg)